build-farm/platform-specific-configurations/linux.sh

```diff
@@ -208,16 +221,32 @@ if [ "${ARCHITECTURE}" == "riscv64" ] && [ "$(uname -m)" == "x86_64" ]; then
       exit 1
     fi
   fi
+
+  # RISC-V cross compile settings for all VARIANT values
   echo RISC-V cross-compilation setup ...  Setting RISCV64, LD_LIBRARY_PATH, PATH, CC, CXX
   export RISCV64=/opt/riscv_toolchain_linux
   export LD_LIBRARY_PATH=$RISCV64/lib64
   if [ "${VARIANT}" == "${BUILD_VARIANT_BISHENG}" ]; then
     export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:$BUILD_LIBRARY_PATH
   fi
-  export PATH="$RISCV64/bin:$PATH"
-  export CC=$RISCV64/bin/riscv64-unknown-linux-gnu-gcc
-  export CXX=$RISCV64/bin/riscv64-unknown-linux-gnu-g++
-  CONFIGURE_ARGS_FOR_ANY_PLATFORM="${CONFIGURE_ARGS_FOR_ANY_PLATFORM} --openjdk-target=riscv64-unknown-linux-gnu --with-sysroot=/opt/fedora28_riscv_root --with-boot-jdk=$JDK_BOOT_DIR"
+
+  if [ -r "$RISCV64/bin/riscv64-unknown-linux-gnu-g++" ]; then
```

Comment:
what do we do if the if or elif aren't met, will this fail?

Reply:
The OpenJ9 build process will use a default - I haven't tested for other situations, but Iv'e been testing so many conditions I don't think it's worth my time to check all the others for this to answer what might happen in all possible scenarios. It would also depend on the users `PATH` and/or `CC`/`CXX` settings.